hilti/runtime/src/types/stream.cc

```diff
@@ -59,6 +59,9 @@ const Chunk* Chain::findChunk(const Offset& offset, const Chunk* hint_prev) cons
 
     const Chunk* c = _head.get();
 
+    if ( ! hint_prev )
+        hint_prev = _tail; // see if null means last chunk
```

Comment:
This is pretty subtle, so let's expand the comment a bit, maybe
```suggestion
        // A very common way this function gets called without `hint_prev` is
    // `Stream::unsafeEnd` via `Chain::unsafeEnd` in construction of an
    // `UnsafeConstIterator` from a `SafeConstIterator`; in this case the chunk
    // for `end()` will be `nullptr`. Optimize for that case by assuming we
    // always want a chunk near the end if no hint is given.
    if ( ! hint_prev )
        hint_prev = _tail;
```